examples/sample-cmd/main.go

```diff
@@ -23,6 +25,36 @@ func main() {
 		return fmt.Sprintf("Hello %s!", c.Param("name")), nil
 	})
 
+	app.SubCommand("spinner", func(ctx *gofr.Context) (interface{}, error) {
+		// initialize the spinner and defer stop it
+		defer terminal.NewDotSpinner(ctx.Out).Spin().Stop()
+
+		// stimulate a time-taking process
+		time.Sleep(2 * time.Second)
+
+		return "Process Complete", nil
+	})
+
+	app.SubCommand("progress", func(ctx *gofr.Context) (interface{}, error) {
+		p := terminal.NewProgressBar(ctx.Out, 100)
+
+		for i := 1; i <= 100; i++ {
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			default:
+				// do a time taking process or compute a small subset of a bigger problem,
+				// this could be processing batches of a data set.
+				time.Sleep(50 * time.Millisecond)
+
+				// increment the progress to display on the progress bar.
+				p.Incr(int64(1))
+			}
```

Comment:
This is the way to do, otherwise the time.Sleep is still blocking

```suggestion
			select {
			case <-ctx.Done():
				return nil, ctx.Err()
			case <-time.After(50 * time.Millisecond):
				// do a time taking process or compute a small subset of a bigger problem,
				// this could be processing batches of a data set.

				// increment the progress to display on the progress bar.
				p.Incr(int64(1))
			}
```